xcmailr-webapp/src/main/java/controllers/restapi/MailApiController.java

```diff
@@ -118,7 +119,7 @@ public Result listMails(@Param("mailboxAddress") @Email final String mailboxAddr
      * @return the details of the mail as a {@link MailData} object in JSON format
      */
     // @Get("/api/v1/mails/<mailId>")
-    public Result getMail(@PathParam("mailId") @DbId final Long mailId, @Attribute("userId") @DbId final Long userId,
+    public Result getMail(@PathParam("mailId") @DbId final Optional<Long> mailId, @Attribute("userId") @DbId final Long userId,
```

Comment:
The parameter `mailId` is not optional.

Reply:
Reverted to non-optional parameter. However, this prevents us to perform custom error handling in case `mailId` is invalid (could not be parsed as Long)